### ISSUE TITLE
Appropriate xml parse library

### ISSUE THREAD

**Issue by programmerr47** [programmerr47/rss-simple-app#7]
Need to find appropriate java library for parsing xml to pojos. 
There is JAXB, but could be some alternatives. Need to choose.

**Comment by programmerr47** [comment]
Started to investigating firstly about alternatives and didn't found good one for JAXB and as handy as JAXB is. But I faced with [this article](https://www.javacodegeeks.com/2011/12/jaxb-sax-dom-performance.html). Here author compare JAXB with other more verbose parsers (e.t.c. SAX).  

So JAXB is more neat, but user pays performance for that. SAX is more fast than JAXB. But it really matters only for big data. For small information JAXB works no more than for 50ms, where SAX need to take near 5ms to parse xml file. Ok it is much smaller and for android we need to make process very fast for increase responsibility of application. But I think user will not notice real difference between 50ms and 5ms. Especially when we have additionally network connection that needs from 100ms and above. And of course in future we will have a cache for data, so in this case performance difference become less observable than distinction in code 
conciseness, readability and maintainability.

Also need to investigate ability to combine parsing xml with JAXB and building pojos with Builder pattern to provide immutable model. It can be more verbose at first time. But [lombok](https://projectlombok.org/) will solve that problem.

**Comment by programmerr47** [comment]
After some research I came to pre-conclusion that there is no way (easy way) to use JAXB in conjunction with Builder pattern. Okey, we can use XmlAdapter for it, but it will impose more work to be done just for dobious code beauty. Also with this approach code will be more verbose, so it will be better to use SAX in this case. 

But! Still we can unmarshal xml to immutable objects. To do it we need to add explicit private no-args-constructor. It will clutter up the code little bit, but it more concise and fast solution that will not affect "interface" for "user" of this class.

Here is a bunch of articles to do that: [this](http://aniketshaligram.blogspot.ru/2010/05/jaxb-immutable-objects.html), [this](http://stackoverflow.com/questions/6342387/best-approach-for-marshaling-unmarshaling-immutable-objects-with-builder-pattern) and [this](http://stackoverflow.com/questions/11030805/creating-immutable-objects-usingjaxb).

I will close this issue, since we found appropriate for our needs solution.
